src/wallet/spend.rs

```diff
@@ -413,6 +523,25 @@ impl Wallet {
             total_tolerance
         );
 
+        // The actual fee is the difference between the sum of output amounts from the total input amount
+        let actual_fee = total_input_value
+            - (tx.output.iter().fold(Amount::ZERO, |a, txo| {
+                a.checked_add(txo.value)
+                    .expect("output amount summation overflowed")
+            }));
+
+        let tx_size = tx.weight().to_vbytes_ceil();
+        // Note : The feerates are sats/vbyte
+        let actual_feerate = actual_fee.to_sat() as f32 / tx_size as f32;
```

Comment:
Keep stuff clean, and handle the expect block
```suggestion
       let total_output_value = tx.output.iter()
    .map(|txo| txo.value)
    .try_fold(Amount::ZERO, |acc, val| acc.checked_add(val))
    .expect("output amount summation overflowed");

      let actual_fee = total_input_value - total_output_value;
      let tx_size = tx.weight().to_vbytes_ceil();
      let actual_feerate = actual_fee.to_sat() as f32 / tx_size as f32;
```